dashboard/app/models/user_permission.rb

```diff
@@ -46,6 +46,8 @@ class UserPermission < ApplicationRecord
     UNIVERSAL_INSTRUCTOR = 'universal_instructor'.freeze,
     # Grants access to use AI Tutor which uses AI Chat API
     AI_TUTOR_ACCESS = 'ai_tutor_access'.freeze,
+    #  Grants access to an internal tool to pull AI-evaluated samples of student work
+    DATASET_MAKER = 'dataset_maker'.freeze,
```

Comment:
can the name please be more specific, at least enough to disambiguate for example from applab datasets and possibly emphasize they will have access to student code? I would nominate `STUDENT_CODE_DATASET_MAKER` or even `STUDENT_CODE_ACCESS` with the latter perhaps being more to the point about what the grantee will have access to.

Reply:
`STUDENT_WORK_ACCESS` sounds great to me! #naming